### PR TITLE
Fix remote db working

### PR DESCRIPTION
**Pull request by mrrobot47**
Closes https://github.com/EasyEngine/site-type-wp/issues/42

### CODE REVIEW

File: src/WordPress.php [hunk 1]
```diff
@@ -476,35 +476,73 @@ private function generate_default_conf( $site_type, $cache_type, $server_name )
...
-			$launch = \EE::launch( sprintf( "docker network inspect %s --format='{{ (index .IPAM.Config 0).Gateway }}'", $this->site_data['site_url'] ) );
+		if ( '127.0.0.1' === $db_host || 'localhost' === $db_host ) {
+			$launch = \EE::launch( sprintf( "docker exec %s bash -c \"ip route show default | cut -d' ' -f3\"", $container_name ) );
 
 			if ( ! $launch->return_code ) {
-				$this->site_data['db_host'] = trim( $launch->stdout, "\n" );
+				$db_host = trim( $launch->stdout, "\n" );
 			} else {
-				throw new \Exception( 'There was a problem inspecting network. Please check the logs' );
+				throw new \Exception( 'There was a problem connecting to connection test container. Please check the logs' );
```
Comment (mbtamuli) [inline src/WordPress.php]: Confusing message.

File: src/WordPress.php [hunk 2]
```diff
@@ -476,35 +476,73 @@ private function generate_default_conf( $site_type, $cache_type, $server_name )
...
+			$check_tables = sprintf( "docker exec %s bash -c \"mysql --host='%s' --port='%s' --user='%s' --password='%s' --execute='USE %s; show tables;'\"", $container_name, $db_host, $this->site_data['db_port'], $this->site_data['db_user'], $this->site_data['db_password'], $this->site_data['db_name'] );
+
+			$launch = \EE::launch( $check_tables );
+			if ( ! $launch->return_code ) {
+				$tables = trim( $launch->stdout, "\n" );
+				if ( ! empty( $tables ) ) {
+					throw new \Exception( sprintf( 'Some database tables seem to exist in database %s. Please backup and reset the database or use `--force` in the site create command to reset it.', $this->site_data['db_name'] ) );
+				}
+			} else {
+				throw new \Exception( 'There was a problem in connecting and the database. Please check the logs' );
```
Comment (mbtamuli) [inline src/WordPress.php]: Confusing message. Please correct.

File: src/WordPress.php [hunk 3]
```diff
@@ -476,35 +476,79 @@ private function generate_default_conf( $site_type, $cache_type, $server_name )
...
-		if ( in_array( $this->site_data['db_host'], [ 'db', GLOBAL_DB ] ) ) {
+		if ( in_array( $this->site_data['db_host'], [ 'db', GLOBAL_DB ], true ) ) {
 			return;
 		}
+		$db_host        = $this->site_data['db_host'];
+		$img_versions   = \EE\Utils\get_image_versions();
+		$container_name = \EE\Utils\random_password();
+		$network        = ( GLOBAL_DB === $this->site_data['db_host'] ) ? "--network='" . GLOBAL_FRONTEND_NETWORK . "'" : '';
+
+		if ( ! \EE::exec( sprintf( 'docker run --name %s %s -e MYSQL_ROOT_PASSWORD=%s -d --restart always easyengine/mariadb:%s', $container_name, $network, \EE\Utils\random_password(), $img_versions['easyengine/mariadb'] ) ) ) {
```
Comment (rahulsprajapati) [inline src/WordPress.php]: Please move this sprintf to another variable and use it here for better readability. Like you've done below in `$check_db_exists` variable.

File: src/WordPress.php [hunk 4]
```diff
@@ -476,35 +476,79 @@ private function generate_default_conf( $site_type, $cache_type, $server_name )
...
 
-		if ( ! \EE::exec( sprintf( "docker run -it --rm %s easyengine/mariadb:%s sh -c \"mysql --host='%s' --port='%s' --user='%s' --password='%s' --execute='EXIT'\"", $network, $img_versions['easyengine/mariadb'], $this->site_data['db_host'], $this->site_data['db_port'], $this->site_data['db_user'], $this->site_data['db_password'] ) ) ) {
+		if ( ! \EE::exec( sprintf( "docker exec %s sh -c \"mysql --host='%s' --port='%s' --user='%s' --password='%s' --execute='EXIT'\"", $container_name, $db_host, $this->site_data['db_port'], $this->site_data['db_user'], $this->site_data['db_password'] ) ) ) {
+			\EE::exec( "docker rm -f $container_name" );
 			throw new \Exception( 'Unable to connect to remote db' );
 		}
-
 		\EE::success( 'Connection to remote db verified' );
+
+		$name            = str_replace( '_', '\_', $this->site_data['db_name'] );
```
Comment (rahulsprajapati) [inline src/WordPress.php]: Can you give me reason/example why this is required? I guess for db name => test_db the $name value will be `test\_db` 

> docker exec %s bash -c \"mysqlshow --user='%s' --password='%s' --host='%s' --port='%s' 'test\_db'\"

I'm not sure why this is required

File: src/WordPress.php [hunk 5]
```diff
@@ -476,35 +476,79 @@ private function generate_default_conf( $site_type, $cache_type, $server_name )
...
+			\EE::log( sprintf( 'Database `%s` does not exist. Attempting to create it.', $this->site_data['db_name'] ) );
+			$create_db_command = sprintf( "docker exec %s bash -c \"mysql --host='%s' --port='%s' --user='%s' --password='%s' --execute='CREATE DATABASE %s;'\"", $container_name, $db_host, $this->site_data['db_port'], $this->site_data['db_user'], $this->site_data['db_password'], $this->site_data['db_name'] );
+
+			if ( ! \EE::exec( $create_db_command ) ) {
+				\EE::exec( "docker rm -f $container_name" );
+				throw new \Exception( sprintf( 'Could not create database `%s` on `%s:%s`. Please check if %s has rights to create database or manually create a database and pass with `--dbname` parameter.', $this->site_data['db_name'], $this->site_data['db_host'], $this->site_data['db_port'], $this->site_data['db_user'] ) );
+			}
+		} else {
+			if ( $this->force ) {
+				\EE::exec( sprintf( "docker exec %s bash -c \"mysql --host='%s' --port='%s' --user='%s' --password='%s' --execute='DROP DATABASE %s;'\"", $container_name, $db_host, $this->site_data['db_port'], $this->site_data['db_user'], $this->site_data['db_password'], $this->site_data['db_name'] ) );
```
Comment (rahulsprajapati) [inline src/WordPress.php]: When you've this much to add use multiple lines for readability. 

```php
\EE::exec( 
	sprintf( 
		"docker exec %s bash -c \"mysql --host='%s' --port='%s' --user='%s' --password='%s' --execute='DROP DATABASE %s;'\"",
		$container_name, 
		$db_host, 
		$this->site_data['db_port'],
		$this->site_data['db_user'], 
		$this->site_data['db_password'],
		$this->site_data['db_name'] 
	)
);
```

**Pull request merged**